### PR TITLE
WIP: Decode conflicts

### PR DESCRIPTION
**Pull request by glasserc**
If you use a remote transformer + SERVER_WINS syncing strategy + have a conflict, you can get undecoded records in your database. This seems to fix it for me but doesn't include tests. f? @leplatrem @n1k0 ?

### CODE REVIEW

File: src/collection.js [hunk 1]
```diff
@@ -889,11 +889,15 @@ export default class Collection {
 
         // The result of a batch returns data and permissions.
         // XXX: permissions are ignored currently.
-        const conflicts = synced.conflicts.map(({type, local, remote}) => {
+        const conflicts = Promise.all(synced.conflicts.map(({type, local, remote}) => {
           // Note: we ensure that local data are actually available, as they may
           // be missing in the case of a published deletion.
-          return {type, local: local && local.data || {}, remote};
-        });
+          return this._decodeRecord("remote", local || {}).then(realLocal => {
```
Comment (leplatrem) [inline src/collection.js]: Do we really need to decode the local one?

Comment (n1k0) [inline src/collection.js]: I share the same question :) Maybe writing a few tests would help figuring this out!

**Comment by glasserc** [comment]
I've written a couple tests in #525. I pushed them as a separate branch/separate WIP PR because this fix seems to break a bunch of other stuff.

**Comment by glasserc** [comment]
Closing this in favor of #525, which has the correct fix, correctly copied out of the other source tree that I got working.

**Pull request closed without merging**